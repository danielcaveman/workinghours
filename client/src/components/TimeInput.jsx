import React from "react";
import styled from "styled-components";

const TimeInputContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

const Label = styled.label`
  font-size: 1.5rem;
`;

const Time = styled.input`
  border: 0;
  outline: 0;
  border-bottom: 2px solid;
  border-bottom: 2px solid #5eabf2;

  &::-webkit-calendar-picker-indicator {
    background: none;
    display: none;
  }
`;

function TimeInput({ value, onChange, name, label }) {
  return (
    <TimeInputContainer>
      <Label htmlFor={name}>{label}</Label>
      <Time
        type="time"
        onChange={onChange}
        value={value || ""}
        id={name}
        name={name}
      />
    </TimeInputContainer>
  );
}

export default TimeInput;
