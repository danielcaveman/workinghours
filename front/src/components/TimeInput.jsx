import styled from "styled-components";

const Time = styled.input`
  border: 0;
  outline: 0;
  border-bottom: 2px solid;
  border-bottom: 2px solid #5eabf2;

  &:disabled {
    border-bottom: 2px solid #cccccc;
  }

  &::-webkit-calendar-picker-indicator {
    background: none;
    display: none;
  }
`;

function TimeInput({ condition, value, setValue, name }) {
  return (
    <Time
      type="time"
      disabled={!condition}
      value={value || ""}
      onChange={(e) => setValue(e.target.value)}
      id={name}
      name={name}
    />
  );
}

export default TimeInput;
