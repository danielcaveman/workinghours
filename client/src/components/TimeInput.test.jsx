import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TimeInput from "./TimeInput";

describe("<TimeInput />", () => {
  it("Should have a value", () => {
    const fn = jest.fn();
    render(<TimeInput onChange={fn} inputValue="13:58" label={"Date"} />);
    const input = screen.getByLabelText("Date");

    expect(input.value).toBe("13:58");
  });

  it("Should call onChange function on each key pressed", () => {
    const fn = jest.fn();
    render(<TimeInput onChange={fn} label={"Date"} />);
    const input = screen.getByLabelText("Date");

    const value = "13:58";
    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(2);
  });
});
