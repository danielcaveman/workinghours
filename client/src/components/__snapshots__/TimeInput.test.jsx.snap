// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeInput /> Should match snapshot 1`] = `
<div>
  <input
    aria-label="Date"
    class="sc-bdfBwQ kOyfgn"
    type="time"
    value=""
  />
</div>
`;
